### PR TITLE
feat: child hydration before hydration of main element

### PR DESCRIPTION
**Pull request by supproduction**
fixes hydration flickering 

closes [HRZ-2614](https://spryker.atlassian.net/browse/HRZ-2614)

[HRZ-2614]: https://spryker.atlassian.net/browse/HRZ-2614?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/base/utilities/src/decorators/hydratable/hydratable.ts [hunk 1]
```diff
@@ -159,18 +161,18 @@ function hydratableClass<T extends Type<HTMLElement>>(
       const states = this[asyncStates];
 
       setTimeout(() => {
```
Comment (dunqan) [inline libs/base/utilities/src/decorators/hydratable/hydratable.ts]: all this logic should only happen on clientside, yes?

File: libs/base/ui/structure/text/src/text.component.ts [hunk 2]
```diff
@@ -39,14 +64,10 @@ export class TextComponent extends LitElement implements TextProperties {
...
 
-  protected get shouldBeTruncated(): boolean {
-    return !!Number(getComputedStyle(this).getPropertyValue('--line-clamp'));
-  }
-
   protected override render(): TemplateResult {
     return html`
       <div ${ref(this.containerRef)}>
-        <slot @slotchange=${this.setup.bind(this)}></slot>
+        <slot></slot>
```
Comment (dunqan) [inline libs/base/ui/structure/text/src/text.component.ts:60]: We dropped reacting on `slotchange` event here. Are we safe? maybe @tobi-or-not-tobi can you take a look at this one?

File: libs/base/ui/structure/text/src/text.component.ts [hunk 3]
```diff
@@ -10,27 +10,52 @@ import { truncateFix } from './prehydrate';
...
+    }
+
+    super.willUpdate(changedProperties);
+  }
+
+  protected updated(changedProperties: PropertyValues): void {
+    if (
+      changedProperties.has('truncateAfter') &&
+      this.truncateAfter &&
+      !this.resizeObserver
```
Comment (tobi-or-not-tobi) [inline libs/base/ui/structure/text/src/text.component.ts]: let's leave this to the setup method maybe?

File: libs/base/ui/structure/text/src/text.component.ts (hunk 2 above)
Comment (tobi-or-not-tobi) [reply libs/base/ui/structure/text/src/text.component.ts:60]: yes, we have an alternative implementation using the lifecycle method

File: libs/base/utilities/src/decorators/hydratable/hydratable.ts (hunk 1 above)
Comment (supproduction) [reply libs/base/utilities/src/decorators/hydratable/hydratable.ts]: we don't trigger this on server side. But agree

File: libs/base/ui/structure/text/src/text.component.ts (hunk 3 above)
Comment (tobi-or-not-tobi) [reply libs/base/ui/structure/text/src/text.component.ts]:
```
if (!this.truncateAfter || !this.resizeObserver) this.truncation = false;
```

File: libs/base/ui/structure/text/src/text.component.ts [hunk 4]
```diff
@@ -10,27 +10,52 @@ import { truncateFix } from './prehydrate';
...
+  protected resizeObserver?: ResizeObserver;
+
+  protected willUpdate(changedProperties: PropertyValues): void {
+    if (changedProperties.has('truncateAfter')) {
+      if (this.truncateAfter) {
+        this.style.setProperty('--line-clamp', String(this.truncateAfter));
+      } else {
+        this.style.removeProperty('--line-clamp');
+      }
+    }
```
Comment (tobi-or-not-tobi) [inline libs/base/ui/structure/text/src/text.component.ts]: nitpick: we could simplify as `this.style.removeProperty('--line-clamp', null);` should remove the property afaik

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**